src/hooks/useAuth/useAuth.ts

```diff
@@ -0,0 +1,30 @@
+import { useContext } from "react";
+import AuthContext from "context/AuthContext";
+import { GC_API_TOKEN } from "config/localStorageKeys";
+
+export interface Auth {
+  isAuthenticated: boolean;
+  login: () => void;
+  logout: () => void;
+}
+const useAuth = (): Auth => {
+  const { isAuthenticated, setIsAuthenticated } = useContext(AuthContext);
+
+  const login = () => {
```

Comment:
Since those methods _will_ be async, you may as well set them as so now

Also, should the hook maybe expose an error ?

e.g

```javascript
  return {
    isAuthenticated,
    authError, // defaults to null
    login,
    logout,
  };
```